File: dotnet/lab1/test/Cegeka.Inbox.Tests/InboxTest.cs
using System.Collections.Generic;

namespace Cegeka.Inbox.Tests;

public class InboxTest
{
    private readonly MessageSenderForTest messageSender = new();

    public InboxTest()
    {
        MessageSender.SetInstance(messageSender);
    }

    [Fact]
    public void WhenMessageArrivesThenSendMessageToReception()
    {
        var inbox = new Inbox();

        inbox.ReceiveMessage("Ser Pounce", "Hello Cegeka!");

        Assert.Equal(new Dictionary<string, string>
        {
            ["reception"] = "Hello Cegeka!"
        }, messageSender.Messages);
    }

    private sealed class MessageSenderForTest : MessageSender
    {
        public Dictionary<string, string> Messages { get; } = new();

        protected override void AcceptMessage(string to, string message)
        {
            Messages[to] = message;
        }
    }
}

File: java/lab3/src/test/java/be/cegeka/inbox/InboxTest.java
package be.cegeka.inbox;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.Map;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.data.MapEntry.entry;

public class InboxTest {

    private final MessageSenderForTest messageSender = new MessageSenderForTest();

    @BeforeEach
    void setUp() {
        MessageSender.setInstance(messageSender);
    }

    @Test
    void whenMessageArrivesThenSendMessageToReception() {
        Inbox inbox = new Inbox();

        inbox.receiveMessage("Ser Pounce", "Hello Cegeka!");

        assertThat(messageSender.messages).containsExactly(entry("reception", "Hello Cegeka!"));
    }

    private static class MessageSenderForTest extends MessageSender {
        Map<String, String> messages = new HashMap<>();

        void acceptMessage(String to, String message) {
            messages.put(to, message);
        }
    }
}
